### PR TITLE
Unify impl blocks by wrapping them into a div

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
The blanket and "auto traits" sections are wrapped into a `div` with an ID. This PR fixes this incoherence by wrapping each impl section (the "deref impl" and the "inherent impl" sections were missing it). It'll also make some tests simpler to write.

r? @notriddle

### CODE REVIEW

File: src/librustdoc/html/render/mod.rs [hunk 1]
```diff
@@ -1115,7 +1119,9 @@ fn render_assoc_items_inner(
         }
         if !impls_buf.is_empty() {
             w.push_buffer(tmp_buf);
+            write!(w, "<div id=\"{}-list\">", id);
```
Comment (notriddle) [inline src/librustdoc/html/render/mod.rs:1122]: I see that you manually added `implementations-list` to the id_map, and the ID of the deref H1 items gets added to the id_map by `derive_id`. But I can't find anywhere that the modified ID. where you take a deref version and add `-list` to the end, that gets added to the id_map.

It seems like I could engineer an ID conflict by adding `# Deref Methods U8 List` as a header to my markdown.

Comment (GuillaumeGomez) [reply src/librustdoc/html/render/mod.rs:1122]: Good point! I'll add a `derive_id` for it.

**Comment by GuillaumeGomez** [comment]
I added the missing `derive_id`.

**Review by notriddle — APPROVED** [review]

**Pull request merged**